front/temporal/labs/activities.ts

```diff
@@ -0,0 +1,313 @@
+    );
+    return;
+  }
+
+  const convRes = await dust.createConversation({
+    title: null,
+    visibility: "unlisted",
+    message: {
+      content: transcriptContent as string,
+      mentions: [{ configurationId }],
+      context: {
+        timezone: "Europe/Paris",
+        username: "labs-transcript-processor",
+        fullName: null,
+        email: null,
+        profilePictureUrl: null,
+      },
+    },
+    contentFragment: undefined,
+  });
+  if (convRes.isErr()) {
+    logger.error(
+      "[processGoogleDriveTranscriptActivity] Error creating conversation",
+      convRes.error
+    );
+    return new Err(new Error(convRes.error.message));
+  } else {
+    conversation = convRes.value.conversation;
+    userMessage = convRes.value.message;
+
+    logger.info(
+      "[processGoogleDriveTranscriptActivity] Created conversation " +
+        conversation.sId
+    );
+  }
+
+  if (!conversation) {
+    logger.error(
+      "[processGoogleDriveTranscriptActivity] No conversation found. Stopping."
+    );
+    return;
+  }
+
+  const agentMessages = conversation.content
+    .map((versions: any) => {
+      const m = versions[versions.length - 1];
+      return m;
+    })
+    .filter((m) => {
+      return (
+        m &&
+        m.type === "agent_message" &&
+        m.parentMessageId === userMessage?.sId
+      );
+    });
+
+  if (agentMessages.length === 0) {
+    return new Err(new Error("Failed to retrieve agent message"));
+  }
+  const agentMessage = agentMessages[0] as AgentMessageType;
+
+  const streamRes = await dust.streamAgentMessageEvents({
+    conversation: conversation,
+    message: agentMessage,
+  });
+
+  if (streamRes.isErr()) {
+    return new Err(new Error(streamRes.error.message));
+  }
+
+  let fullAnswer = "";
+  for await (const event of streamRes.value.eventStream) {
+    switch (event.type) {
+      case "user_message_error": {
+        return new Err(
+          new Error(
+            `User message error: code: ${event.error.code} message: ${event.error.message}`
+          )
+        );
+      }
+
+      case "agent_error": {
+        return new Err(
+          new Error(
+            `Agent message error: code: ${event.error.code} message: ${event.error.message}`
+          )
+        );
+      }
+      case "generation_tokens": {
+        fullAnswer += event.text;
+        break;
+      }
+
+      default:
+      // Nothing to do on unsupported events
+    }
+  }
+
+  // SEND EMAIL WITH SUMMARY
+  const msg = {
+    to: transcriptsConfiguration.emailToNotify || user.email,
+    from: "team@dust.tt",
+    subject: `[DUST] Meeting summary - ${transcriptTitle}`,
+    html: `${fullAnswer}<br>` + `The Dust team`,
+  };
+  void sgMail.send(msg).then(() => {
+    logger.info(
+      "[processGoogleDriveTranscriptActivity] Email sent with summary"
+    );
+  });
+
+  // CREATE HISTORY RECORD
+  LabsTranscriptsHistoryResource.makeNew({
```

Comment:
You might want to try to fetch the transcript before so this activity can be fully idempotent and avoid the thenable.